fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Modifier.kt

```diff
@@ -49,3 +56,30 @@ fun Modifier.dashedBorder(
         )
     },
 )
+
+/**
+ * Used when clickable needs to be debounced to prevent rapid successive clicks
+ * from calling the onClick function.
+ *
+ * @param debounceInterval The length of time to wait between click events in milliseconds
+ * @param onClick Callback for when item this modifier effects is clicked
+ */
+fun Modifier.debouncedClickable(
+    debounceInterval: Long = 1000L,
+    onClick: () -> Unit,
+) = composed(
+    factory = {
+        var lastClickTime: Long by remember { mutableStateOf(0) }
+
+        this.then(
+            Modifier.clickable(
+                onClick = {
+                    if (SystemClock.elapsedRealtime() - lastClickTime > debounceInterval) {
```

Comment:
maybe extract the system clock time to a local value since it could change between this `if` and the `lastClickTime` reassignment?

Reply:
good point, I moved system clock time to a local value